src/emoji/data.js

```diff
@@ -29,6 +29,32 @@ export const getFilteredEmojiNames = (
   query: string,
   activeImageEmojiByName: $ReadOnly<{ [string]: ImageEmojiType }>,
 ): string[] => {
+  // Check if the query itself is an emoji, and is known to Zulip. If
+  // so, return the emoji name in an array. Emoji with single-codepoint and
+  // multi-codepoint have length 2 and 3 respectively.
+  if (query.length === 2) {
+    const emojiName = unicodeEmojiNames.find(
+      key => unicodeCodeByName[key] === query.codePointAt(0).toString(16),
+    );
+    if (emojiName !== undefined) {
+      return [emojiName];
+    }
+  } else if (query.length === 3) {
+    const emojiName = unicodeEmojiNames.find(
+      key =>
+        unicodeCodeByName[key]
+        === `${query
+          .codePointAt(0)
```

Comment:
This indentation looks confusing to me; maybe Prettier and ESLint are making it look like this. One thing that might help is to give it some breathing room by using the curly-brace and `return` statement syntax with the lambda, e.g.,

```javascript
key => {
  // ...
  return /* ... */;
}
```

Then, in there, you could have separate variables corresponding to the first and second code points, just so it's a bit clearer what's going on.